src/InformaticsGateway/Services/Scp/Payload.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright 2021 MONAI Consortium
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Ardalis.GuardClauses;
+using Monai.Deploy.InformaticsGateway.Common;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Monai.Deploy.InformaticsGateway.Services.Scp
+{
+    internal class Payload : List<FileStorageInfo>, IDisposable
```

Comment:
It depends on how you see it; I see Payload as a container of files therefore, it inherits from a LIst<>.  The benefit is that you get all methods from List<> for free.

Reply:
It does look convenient if one considers the fragments of a payload are of the same type, like here, the FileStorageInfo. This convenience will become a liability when payload becomes more complex, e.g. multiple lists that have some relationship, which cannot be encapsulated in a common type.

Since the Payload is internal, and the payloads targeted in the initial release are purely DICOM study/series, this design will work, but I can also see, this will need to change when EHR data becomes part of the payload along with DICOM.

Reply:
There are actually 2 sub-classes.  DicomFieStorageInfo and FhirFileStorageInfo in case you missed them.

Reply:
Yes, I did see it, and the payload itself is just a list of FileStorageInfo objects, DICOM or EHR. So it does to change limitation on the design, but OK for the current use case where payload is limited to a list of input files.